plugins/inputs/azure_storage_queue/azure_storage_queue.go

```diff
@@ -19,108 +19,101 @@ import (
 
-	serviceURL *azqueue.ServiceURL
+	client *azqueue.ServiceClient
 }
 
 func (*AzureStorageQueue) SampleConfig() string {
 	return sampleConfig
 }
 
 func (a *AzureStorageQueue) Init() error {
+	// Check settings
 	if a.StorageAccountName == "" {
 		return errors.New("account_name must be configured")
 	}
 
 	if a.StorageAccountKey == "" {
 		return errors.New("account_key must be configured")
 	}
+
+	// Prepare the client
+	if a.EndpointURL == "" {
+		a.EndpointURL = "https://" + a.StorageAccountName + ".queue.core.windows.net"
+	}
+	credentials, err := azqueue.NewSharedKeyCredential(a.StorageAccountName, a.StorageAccountKey)
+	if err != nil {
+		return fmt.Errorf("creating shared-key credentials failed: %w", err)
+	}
+
+	client, err := azqueue.NewServiceClientWithSharedKeyCredential(a.EndpointURL, credentials, nil)
+	if err != nil {
+		return fmt.Errorf("creating client failed: %w", err)
+	}
+	a.client = client
+
 	return nil
 }
 
 func (a *AzureStorageQueue) Gather(acc telegraf.Accumulator) error {
-	serviceURL, err := a.getServiceURL()
-	if err != nil {
-		return err
-	}
+	ctx := context.Background()
 
-	ctx := context.TODO()
+	a.Log.Debugf("Listing queues of storage account %q", a.StorageAccountName)
 
-	for marker := (azqueue.Marker{}); marker.NotDone(); {
-		a.Log.Debugf("Listing queues of storage account %q", a.StorageAccountName)
-		queuesSegment, err := serviceURL.ListQueuesSegment(ctx, marker,
-			azqueue.ListQueuesSegmentOptions{
-				Detail: azqueue.ListQueuesSegmentDetails{Metadata: false},
-			})
+	// Iterate through the queues and generate metrics
+	pages := a.client.NewListQueuesPager(nil)
+	for pages.More() {
+		response, err := pages.NextPage(ctx)
 		if err != nil {
-			return err
+			return fmt.Errorf("getting next page failed: %w", err)
 		}
-		marker = queuesSegment.NextMarker
 
-		for _, queueItem := range queuesSegment.QueueItems {
-			a.Log.Debugf("Processing queue %q of storage account %q", queueItem.Name, a.StorageAccountName)
-			queueURL := serviceURL.NewQueueURL(queueItem.Name)
-			properties, err := queueURL.GetProperties(ctx)
+		// Get the properties and the message properties for each of the queues
+		for _, queue := range response.Queues {
+			if queue.Name == nil {
+				continue
+			}
+			name := strings.TrimSpace(*queue.Name)
+
+			// Access the queue and get the properties
+			c := a.client.NewQueueClient(*queue.Name)
+			props, err := c.GetProperties(ctx, nil)
 			if err != nil {
-				a.Log.Errorf("Error getting properties for queue %s: %s", queueItem.Name, err.Error())
+				acc.AddError(fmt.Errorf("getting properties for queue %q failed: %w", name, err))
 				continue
 			}
-			var peekedMessage *azqueue.PeekedMessage
+			if props.ApproximateMessagesCount == nil {
+				acc.AddError(fmt.Errorf("unset message count for queue %q", name))
+				continue
+			}
+
+			// Setup the metric elements
+			tags := map[string]string{
+				"account": a.StorageAccountName,
+				"queue":   strings.TrimSpace(name),
```

Comment:
this is already trimmed at line 81, so can be used without trimming.
```suggestion
				"account": a.StorageAccountName,
				"queue":   name,
```